### PR TITLE
Break apart atomcharges tests per type into Mulliken and Lowdin

### PR DESCRIPTION
**Pull request by berquist**
This also includes a fix for old versions of Psi4 where the charge header was inconsistent.

### CODE REVIEW

File: cclib/parser/psi4parser.py [hunk 1]
```diff
@@ -44,6 +45,11 @@ def before_parsing(self):
         # with =>/<= rather than ==>/<==.
         self.subsection = None
 
+        # Regular expression for matching the start of an atomic charges
+        # section; the second format of the header (with square brackets) is
+        # for old Psi4 versions.
+        self.re_atomic_charges_header = re.compile(r"^\s*(\w+) Charges(:?: \(a\.u\.\)| \[a\.u\.\]:)")
```
Comment (langner) [inline cclib/parser/psi4parser.py:51]: Do you think it's better to spell out which ones we expect here, as in `(Mulliken|Lowdin)` ?

File: test/data/testSP.py [hunk 2]
```diff
@@ -68,11 +68,40 @@ def testatomnos(self):
     @skipForLogfile('Molpro/basicMolpro2006', "These tests were run a long time ago and since we don't have access to Molpro 2006 anymore, we can skip this test (it is tested in 2012)")
     @skipForParser('Turbomole','The parser is still being developed so we skip this test')
     def testatomcharges(self):
```
Comment (langner) [inline test/data/testSP.py:70]: Should this then have a Mulliken suffix?

**Review by langner — APPROVED** [review]

File: cclib/parser/psi4parser.py (hunk 1 above)
Comment (berquist) [reply cclib/parser/psi4parser.py:51]: According to https://psicode.org/psi4manual/master/oeprop.html there is only one other type (MBIS), so it is not bad to do `(Mulliken|Lowdin|MBIS)`.

File: test/data/testSP.py (hunk 2 above)
Comment (berquist) [reply test/data/testSP.py:70]: This was intentional, to have `testatomcharges` be generic for all types and `testatomcharges_mulliken` and `testatomcharges_lowdin` to make sure Mulliken and Lowdin are present. Is there a better way of doing this? I can at least improve the docstring to something like `Are any present atomic charges consistent with natom?`.

Comment (amandadumi) [reply test/data/testSP.py:70]: The change in description helped this make sense to me. I see why both the generic and specific tests are needed.

**Review by shivupa — APPROVED** [review]

**Pull request merged**